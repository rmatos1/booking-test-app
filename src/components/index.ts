export * from "./bookingDescription";
export * from "./bookingForm";
export * from "./images";
export * from "./inputGroup";
export * from "./navBar";
export * from "./styledComponents";
export * from "./testComponent";
