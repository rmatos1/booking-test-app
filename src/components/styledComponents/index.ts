export * from './buttons';
export * from './formComponents';
export * from './images';
export * from './layouts';
export * from './lists';
export * from './modalComponents';
export * from './texts';
