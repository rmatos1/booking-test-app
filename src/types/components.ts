import { ReactNode } from "react";

export interface IComponentWithChildren {
  children?: ReactNode;
}

export interface IBaseDrawer extends IComponentWithChildren {
  isDrawerVisible: boolean;
  onCloseDrawer: () => void;
}
