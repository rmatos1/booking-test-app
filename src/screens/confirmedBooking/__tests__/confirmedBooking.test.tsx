import { describe, test } from '@jest/globals';
import { render } from "@testing-library/react";

import { ConfirmedBooking } from "..";
import { TestComponent } from '../../../components';

const setup = (): JSX.Element => {
  return(
    <TestComponent>
      <ConfirmedBooking />
    </TestComponent>
  )
};

describe('<ConfirmedBooking />', () => {
  test('should render correctly', () => {

    const wrapper = render(setup());

    expect(wrapper).toMatchSnapshot();
  });


});
