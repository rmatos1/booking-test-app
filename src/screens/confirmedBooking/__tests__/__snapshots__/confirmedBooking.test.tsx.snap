// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfirmedBooking /> should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style="overflow: unset;"
  >
    <div>
      <nav
        class="sc-fubDmA jSFNkN"
      >
        <a
          aria-current="page"
          class="sc-pGaPU kpubXZ active"
          href="/"
        >
          New Booking
        </a>
        <a
          class="sc-pGaPU zBnL"
          href="/my-bookings"
        >
          My Bookings
        </a>
      </nav>
      <div
        class="sc-bqyKbq gCCXA-D"
      >
        <img
          alt="confirmed image"
          class="sc-eCstZk cgIxof"
          data-testid="image"
        />
        <div
          class="sc-kfzCjt KTkhM"
        >
          <p>
            Sorry, we couldn't identify your booking. However, you can make a new one
          </p>
          <button
            class="sc-jrAGKZ fjUZsQ"
          >
            New Booking
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <nav
      class="sc-fubDmA jSFNkN"
    >
      <a
        aria-current="page"
        class="sc-pGaPU kpubXZ active"
        href="/"
      >
        New Booking
      </a>
      <a
        class="sc-pGaPU zBnL"
        href="/my-bookings"
      >
        My Bookings
      </a>
    </nav>
    <div
      class="sc-bqyKbq gCCXA-D"
    >
      <img
        alt="confirmed image"
        class="sc-eCstZk cgIxof"
        data-testid="image"
      />
      <div
        class="sc-kfzCjt KTkhM"
      >
        <p>
          Sorry, we couldn't identify your booking. However, you can make a new one
        </p>
        <button
          class="sc-jrAGKZ fjUZsQ"
        >
          New Booking
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
